### PR TITLE
Add system type detection

### PR DESCRIPTION
**Pull request by QuinnPainter**
Necessary for games that want to run on both base and color Game Boys and use the GBC's extra features.

### CODE REVIEW

File: inc/sdk/system.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#ifndef GBSDK_SYSTEM_H
+#define GBSDK_SYSTEM_H
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#define CPU_DMG 0x01 // Dot Matrix Game
+#define CPU_CGB 0x11 // Color GameBoy
+#define CPU_MGB 0xFF // Mini GameBoy (Pocket GameBoy)
```
Comment (daid) [inline inc/sdk/system.h:9]: 0xFF is also an SGB2.

File: src/entry.asm [hunk 2]
```diff
@@ -1,5 +1,11 @@
 include "sdk/hardware.inc"
 
+SECTION "gbsdk_systeminfo_ram", WRAM0
```
Comment (daid) [inline src/entry.asm:3]: In theory, if only one of the `CGB` and `SGB` and `DMG` defines is set, you won't need these and can save two bytes of RAM.

Comment (QuinnPainter) [reply src/entry.asm:3]: Well, even if only one of those were set, theoretically someone could want to distinguish between DMG and MGB, or detect a CGB running in DMG mode or something like that. I could certainly add an `IF CGB` around `system_is_gba` though to save a byte.

File: inc/sdk/system.h (hunk 1 above)
Comment (QuinnPainter) [reply inc/sdk/system.h:9]: Right, I can add that. I was just copying that text from hardware,inc.

**Pull request merged**